pkg/sdn/plugin/multitenant.go

```diff
@@ -0,0 +1,126 @@
+package plugin
+
+import (
+	"fmt"
+
+	log "github.com/golang/glog"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/client/cache"
+	kubetypes "k8s.io/kubernetes/pkg/kubelet/container"
+	kerrors "k8s.io/kubernetes/pkg/util/errors"
+	utilwait "k8s.io/kubernetes/pkg/util/wait"
+)
+
+func (node *OsdnNode) MultitenantStartNode() error {
+	if err := node.VnidStartNode(); err != nil {
+		return err
+	}
+	if err := node.SetupEgressNetworkPolicy(); err != nil {
+		return err
+	}
+
+	go utilwait.Forever(node.watchServices, 0)
+	return nil
+}
+
+func (node *OsdnNode) updatePodNetwork(namespace string, oldNetID, netID uint32) error {
```

Comment:
Hm... did I explain that in the commit message? Maybe not... The idea is that there's going to be an openshift-ovs-networkpolicy network plugin soon as well, and it will use the VNID-tracking code, but it won't react to it in the same way. So vnids_node.go = NetNamespace watching, and multitenant.go = openshift-ovs-multitenant-specific policy, and there will later be networkpolicy.go as well.

The fact that updatePodNetwork() gets called directly from vnids_node.go is wrong, yeah, and that's going to change later. (Initially I was planning to *just* move code around in these commits, and not refactor anything, which is why this is like that. Although then I did end up making some code changes too, so maybe I should have fixed this...)